### PR TITLE
ROC-2469 Disable unknown external refernce usage'

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/features/claim/routes/confirmation.ts [hunk 1]
```diff
@@ -3,12 +3,14 @@ import { Paths } from 'claim/paths'
 
 import ClaimStoreClient from 'claims/claimStoreClient'
 import Claim from 'claims/models/claim'
+import OwnershipChecks from 'app/auth/ownershipChecks'
```
Comment (yarpo) [inline src/main/features/claim/routes/confirmation.ts]: @geek-soft didn't we decide we don't use `default exports` in TypeScript?

Comment (yarpo) [reply src/main/features/claim/routes/confirmation.ts]: Why is this not a guard?

File: src/main/features/claim/routes/confirmation.ts [hunk 2]
```diff
@@ -3,12 +3,14 @@ import { Paths } from 'claim/paths'
...
 
 export default express.Router()
   .get(Paths.confirmationPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction) => {
     const { externalId } = req.params
     try {
       const claim: Claim = await ClaimStoreClient.retrieveByExternalId(externalId)
+      OwnershipChecks.checkClaimOwner(res.locals.user, claim)
```
Comment (timja-kainos) [inline src/main/features/claim/routes/confirmation.ts]: This is not the right approach.
`ClaimStoreClient.retrieveByExternalId` should have the check and there need's to be a forbidden page explaining what happened

File: src/main/features/claim/routes/confirmation.ts (hunk 1 above)
Comment (geek-soft) [reply src/main/features/claim/routes/confirmation.ts]: You are right @yarpo , we tend to not use defaults. Please change it @dharmendrak

File: src/main/features/claim/routes/confirmation.ts (hunk 2 above)
Comment (geek-soft) [reply src/main/features/claim/routes/confirmation.ts]: I would just slightly change the approach to not add the ownership check logic to `ClaimStoreClient.retrieveByExternalId` and creating `ClaimStoreService` instead which delegates data retrieval to generic client and throws an error when ownership check failed.

That would keep HTTP client simple with one responsibility only and in sync with the API which does not have any ownership logic.

File: src/main/features/claim/routes/confirmation.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default express.Router()
   .get(Paths.confirmationPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction) => {
     const { externalId } = req.params
     try {
-      const claim: Claim = await ClaimStoreClient.retrieveByExternalId(externalId)
+      const claim: Claim = await ClaimStoreClient.retrieveByExternalId(externalId, res.locals.user.id)
```
Comment (timja-kainos) [inline src/main/features/claim/routes/confirmation.ts]: might be worth some tests for all this

File: src/main/app/claims/claimStoreClient.ts [hunk 4]
```diff
@@ -69,7 +69,11 @@ export default class ClaimStoreClient {
       .get(`${claimStoreApiUrl}/${externalId}`)
       .then(claim => {
         if (claim) {
-          return new Claim().deserialize(claim)
+          if (userId !== claim.submitterId) {
+            throw new Error('You are not allowed to access this resource')
```
Comment (rlewan) [inline src/main/app/claims/claimStoreClient.ts]: Would be nice if this was e.g. a `ForbiddenError` which would map to a forbidden page.

File: src/main/app/claims/claimStoreClient.ts [hunk 5]
```diff
@@ -69,7 +70,11 @@ export default class ClaimStoreClient {
       .get(`${claimStoreApiUrl}/${externalId}`)
       .then(claim => {
         if (claim) {
-          return new Claim().deserialize(claim)
+          if (userId !== claim.submitterId) {
```
Comment (geek-soft) [inline src/main/app/claims/claimStoreClient.ts]: That breaks response journey! Defendant is unable to retrieve claim by external ID

Comment (timja-kainos) [reply src/main/app/claims/claimStoreClient.ts]: oops, :(

Comment (timja-kainos) [reply src/main/app/claims/claimStoreClient.ts]: how did the checks pass if it breaks them?

Comment (pragnesh-patel) [reply src/main/app/claims/claimStoreClient.ts]: I was thinking how the checks passed. How???

Comment (timja-kainos) [reply src/main/app/claims/claimStoreClient.ts]: @rlewan is investigating

**Pull request merged**